Makefile

```diff
@@ -285,4 +286,23 @@ vendor:
 install-go-tools:
 	go install ${GOTESTSUM}
 
-.PHONY: admin.kubeconfig aks.kubeconfig aro az ci-rp ci-clean clean client deploy dev-config.yaml discoverycache fix-macos-vendor generate image-aro-multistage image-fluentbit image-proxy init-contrib lint-go runlocal-rp proxy publish-image-aro-multistage publish-image-fluentbit publish-image-proxy secrets secrets-update e2e.test tunnel test-e2e test-go test-python vendor build-all validate-go unit-test-go coverage-go validate-fips install-go-tools
+ansible-image:
+	docker image exists aro-ansible:$(VERSION) || docker build . -f Dockerfile.ansible --build-arg REGISTRY=$(REGISTRY) --build-arg VERSION=$(VERSION) --no-cache=$(NO_CACHE) --tag aro-ansible:$(VERSION)
```

Comment:
If we want to recreate the aro-ansible container during development, does this mean we will have to manually delete the image? `podman image rm aro-ansiable:whatever-version`

Reply:
Good call on this, +1, We should leverage proper build context and docker caching instead of using `docker image exists`.

This means that when we rebuild without changes, then we hit the cache 100% as long as our docker build context, mostly managed by Dockerfile.ansible.dockerignore (and missing in this PR) is stood up correctly. It also means if we rebuild _with_ changes, we correctly rebuild the image as desired.

Reply:
On my machine, rebuilding the container image takes 5m40s even if all layers end up being the same, mostly due to re-running `pip` to install `ansible` and `azure-cli`. This proved frustrating when all I wanted to do was to run the existing image again. Is there a way of making `docker build` know it's going to be a no-op because nothing (local) has changed?

I'll add an empty `dockerignore` (only the `/ansible` directory gets copied into the container so there's nothing to ignore right now).

Reply:
A manual way that comes to mind, is looping over the files in the image with bash conditional expression `-nt` to determine if they have been modified after the image was built. [Bash Conditional Expressions](https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html), although something native to docker builld or podman builbah would be ideal.

Reply:
I don't see much of the changes to be happening in the docker file or image over a period of time. Would it be a good idea to build a version of the image and make it readily available in a image registry (or as a tar file) so there's not much time wasted on rebuilding it ?